src/Build/Evaluation/Evaluator.cs

```diff
@@ -243,6 +244,11 @@ internal class Evaluator<P, I, M, D>
 #endif
 
         /// <summary>
+        /// The search paths are machine specific and should not change during builds
+        /// </summary>
+        private static readonly EngineFileUtilities.IOCache _fallbackSearchPathsCache = new EngineFileUtilities.IOCache();
```

Comment:
Talked about this with @AndyGerlicher on Friday. We're a bit concerned about this cache having process lifetime. VS is long-lived and someone could conceivably install things while it's opened.

Unfortunately, I don't know how to reduce this concern.